src/utils/account/account.service.ts

```diff
@@ -0,0 +1,38 @@
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config';
+import { UserInput } from 'src/domain/user/user.dto';
+import { IServiceConfig } from 'src/interfaces/service.config.interface';
+import { MsGraphService } from '../ms-graph/ms-graph.service';
+
+@Injectable()
+export class AccountService {
+  constructor(
+    private configService: ConfigService,
+    private msGraphService: MsGraphService
+  ) {}
+
+  authenticationEnabled(): boolean {
+    if (
+      this.configService.get<IServiceConfig>('service')
+        ?.authenticationEnabled === 'false'
+    )
+      return false;
+    return true;
+  }
+
+  async accountUsageEnabled(): Promise<boolean> {
+    // Currently the only factor for account usage is authentication enabled or not i.e. accounts are used if auth is enabled
+    return this.authenticationEnabled();
+  }
+
+  async accountExists(email: string): Promise<boolean> {
+    // If authentication is disabled then the logic should continue as if an account is present
+    if (!this.authenticationEnabled()) return true;
+    return await this.msGraphService.userExists(undefined, email);
```

Comment:
I doubt this will work. MS Graph Service relies on OBO flow, which relies on an access token.